src/common/snippets/src/runtime_configurator.cpp

```diff
@@ -57,6 +75,22 @@ void RuntimeConfigurator::initialization(const lowered::LinearIRCPtr& linear_ir)
     m_latest_shapes.resize(m_io_num);
     m_config->io_data_offsets.resize(m_io_num);
     m_config->tile_rank = linear_ir->get_config().m_loop_depth;
+
+    // InnerSplittedLoops should be inited after OuterSplittedLoops
+    m_ordered_loop_ids.clear();
+    size_t offset = 0;
+    const auto& loop_map = linear_ir->get_loop_manager()->get_map();
+    for (const auto& p : loop_map) {
```

Comment:
I tried to implement this interface `apply` but seems like it depends on type of LoopInfo also.
See:
After `InsertSpecificLoopIterations` and `SplitLoops` we have:
Outer splitted loop - it's `ExpandedLoopInfo0` which have pointer `UnifiedLoopInfo0`.
Inner splitted loop - it's  `ExpandedLoopInfo1` which have pointer `InnerSplittedUnifiedLoopInfo0` which have pointer to `ExpandedLoopInfo0`. It's needed to be implemented so because outer `ExpandedLoopInfo0` may have different increments for each Loop iteration. It means that `InnerSplittedUnifiedLoopInfo0` should points to the corresponding outer `ExpandedLoopInfo`.

Then if we implement apply for `UnifiedLoopInfo` initialization in the method `update_loop_info` and call it for `InnerSplittedUnifiedLoopInfo0`, we will get exception:
```
    LoopInfoSet updated_unified_loops;
    auto update_unified_loop = [&initializated_info_map](const LoopInfoPtr& loop_info) {
        const auto& unified_loop_info = ov::as_type_ptr<lowered::UnifiedLoopInfo>(loop_info);
        OPENVINO_ASSERT(unified_loop_info);
        if (initializated_info_map.count(unified_loop_info) == 0) {
            lowered::pass::InitLoops::update_runtime_parameters(unified_loop_info);
            initializated_info_map[unified_loop_info] = compute_runtime_params(unified_loop_info);
        }
    };

    ....
   // in loop by loop_map
   const auto& current_unified_loop_info = expanded_loop_info->get_unified_loop_info();
   current_unified_loop_info->apply(update_unified_loop, updated_unified_loops);
```

Firstly we call `apply` for  `InnerSplittedUnifiedLoopInfo0` where we call `apply` again for `m_outer_splitted_loop = ExpandedLoopInfo0`. Then We will get exception in applier `update_unified_loop` whecn check that it's really `UnifiedLoopInfo` (because it's not so).

Thus, at first glance it doesn't look safety because we want to use `apply` recurrently for stored loop info which might be of different Loop Info type.
For sure, we can handle it in `applier`. For example, 
```
    auto update_unified_loop = [&initializated_info_map](const LoopInfoPtr& loop_info) {
        const auto& unified_loop_info = ov::as_type_ptr<lowered::UnifiedLoopInfo>(loop_info);
       if (unified_loop_info ) {
           if (initializated_info_map.count(unified_loop_info) == 0) {
                lowered::pass::InitLoops::update_runtime_parameters(unified_loop_info);
                initializated_info_map[unified_loop_info] = compute_runtime_params(unified_loop_info);
            }
        }
    };
```

But what will be if we something goes wrong 🤔 Probably it will be very difficult to debug.
Need to think 🤔

Reply:
Let's continue offline

Reply:
Decided to try to apply the changes in this PR

Reply:
Implemented virtual method `apply(const std::function<void(const LoopInfoPtr&)>& func, LoopInfoSet& applied_loops)`